src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/ExperimentOperator.java

```diff
@@ -62,27 +71,91 @@ public void startExperiment(Experiment experiment) {
+                    log.fatal("Failed to unpublish experiment " + experiment + " on platform " + population.getPlatformId(), e);
+                }
+            }
+
+            experimentsPlatformOperations.setGlobalPlatformStatus(experiment, ExperimentsPlatformStatusPlatformStatus.shutdown);
+            ShutdownRunner runner = new ShutdownRunner(experiment);
+            runner.run();
+        }else if(!isRunning){
+            log.info("Experiment "+experiment.getId()+" is not running ");
+        }else {
+            log.info("Experiment "+experiment.getId()+" is already shutting down");
 
-        try { //TODO Introduce a more elaborate solution - W. Churchill
-            TimeUnit.HOURS.sleep(2);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
         }
+    }
+
+    private void recoverExperimentShutdown(int experimentID){
+        Experiment experiment = experimentFetcher.fetchExperiment(experimentID);
+        ExperimentsPlatformStatusRecord experimentsPlatformStatusRecord = experimentsPlatformOperations.getExperimentsPlatformStatusRecord(experimentID);
+        long passedTime = Timestamp.valueOf(LocalDateTime.now()).getTime()-experimentsPlatformStatusRecord.getTimestamp().getTime();
+        long passedMins = TimeUnit.MILLISECONDS.toMinutes(passedTime);
 
-        EventManager.EXPERIMENT_CHANGE.emit(new ChangeEvent<>(experiment,experiment.toBuilder().setState(Experiment.State.STOPPED).build()));
+        ShutdownRunner runner = new ShutdownRunner(experiment);
+        if(passedMins >= 120){
+            runner.runRemaining(1);
+        }else {
+            runner.runRemaining(120 - (int)passedMins);
+        }
+
+    }
+
+
+    /**
+     * Private class for non-blocking shutdown-process
+     */
+    private class ShutdownRunner extends Thread{
+        private Experiment experiment;
+        private int minutesToShutdown = 120;
+
+        public ShutdownRunner(Experiment experiment){
+            this.experiment = experiment;
+        }
+
+        /**
+         * Waits a specified amount of time until stopping the experiment and setting its state accordingly
+         */
+        public void run(){
+            try {
+                TimeUnit.MINUTES.sleep(minutesToShutdown);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            experimentsPlatformOperations.setGlobalPlatformStatus(experiment,ExperimentsPlatformStatusPlatformStatus.stopped);
+            EventManager.EXPERIMENT_CHANGE.emit(new ChangeEvent<>(experiment,experimentFetcher.fetchExperiment(experiment.getId())));
+        }
+        /**
+        * Waits the remaining time of the shutdown process until stopping the experiment and setting its state accordingly
+        */
+         public void runRemaining(int minutesToShutdown){ //Used by a recover-method
+            this.minutesToShutdown = minutesToShutdown;
+            run();
```

Comment:
You sure this works ? if you call runRemaining blocking you are calling run blocking which will lock the calling thead for <2h